### DISCUSSION TITLE
Row Level Security and supabase-py, how can I identify which user is calling an insert?

### DISCUSSION THREAD

**Discussion by titogarrido** [supabase/supabase#3479] [Questions]
Hi Folks!

I am calling a python API from  Vue app where an user is logged in using supabase. This Python API was created using FastAPI and I have shared the same JWT secret and algorithm so I can ensure who is calling the API. The problem is when I try to execute an insert using supabase client + anon key I am receiving a RLS error, seems that it can't identify who is calling the insert from the python client.

My RLS is:
```
Enable insert for authenticated users only
(role() = 'authenticated'::text)
```

Is it possible somehow to pass which user is calling the python API using supabase-py? I do have the access_token sent by the javascript client but I have no idea how to share it and use on `supabase.table("projects").insert`

**Comment by rmontgomery429** [comment]
If you have the `access_token` from the client I believe you can call `set_auth` on the client. See https://github.com/supabase-community/gotrue-py/pull/3.

**Reply by OliverSpeir** [reply]
I know this is old, but can you show an example of how this would be used?

for example:

```python
from supabase import create_client

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
```

where would you go from here?

**Reply by wagenrace** [reply]
@OliverSpeir this answer helped me with uploading to tables:
https://github.com/supabase-community/supabase-py/issues/645

```python
from supabase import create_client

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

user_data = supabase.auth.sign_in_with_password(
    {"email": user_name, "password": password}
)
session = user_data.session
supabase.postgrest.auth(
    # verified that this was real
    token=session.access_token
)  # this sets the right "authorization" header on HTTP requests to PostgREST
res = (
    supabase.table("tableName")
    .insert(
        data
    )
    .execute()
)
```

**Reply by mxfng** [reply]
@wagenrace Just wanted to thank you for sharing this, I was really stuck working on my first server side implementation of Supabase client and this got me unstuck. <3

**Reply by krishna-praveen** [reply]
Thanks man, after hours of search i finally found the solution.

**Reply by wagenrace** [reply]
> Thanks man, after hours of search i finally found the solution.

I didn't need it myself after updating the package to the latest version. If you still have the problem maybe report it as an issue

**Reply by john-nhoj** [reply]
@wagenrace , do you mean you didn't have to use `postgrest.auth`?
if so, how did you apply RLS then?

**Reply by wagenrace** [reply]
I didn't need the hack after updating the package. They fixed it 
So using it normally now works

RLS I still applied inside supabase

**Comment by Lotus08** [comment]
I was stuck on this issue for 2 weeks. Thank you so much. This saved me so much time.
